motionplan/plannerOptions.go

```diff
@@ -217,20 +235,47 @@ func (p *plannerOptions) SetMinScore(minScore float64) {
 
 // addPbConstraints will add all constraints from the protobuf constraint specification. This will deal with only the topological
 // constraints. It will return a bool indicating whether there are any to add.
-func (p *plannerOptions) addPbTopoConstraints(from, to spatialmath.Pose, constraints *Constraints) bool {
+func (p *plannerOptions) addPbTopoConstraints(
+	fs referenceframe.FrameSystem,
+	startCfg map[string][]referenceframe.Input,
+	from, to PathStep,
+	constraints *Constraints,
+) (bool, error) {
 	topoConstraints := false
 	for _, linearConstraint := range constraints.GetLinearConstraint() {
 		topoConstraints = true
-		p.addLinearConstraints(from, to, linearConstraint)
+		// TODO RSDK-9224: Our proto for constraints does not allow the specification of which frames should be constrainted relative to
+		// which other frames. If there is only one goal specified, then we assume that the constraint is between the moving and goal frame.
+		err := p.addLinearConstraints(fs, startCfg, from, to, linearConstraint)
+		if err != nil {
+			return false, err
+		}
+	}
+	for _, pseudolinearConstraint := range constraints.GetPseudolinearConstraint() {
+		// pseudolinear constraints
+		err := p.addPseudolinearConstraints(fs, startCfg, from, to, pseudolinearConstraint)
+		if err != nil {
+			return false, err
+		}
 	}
 	for _, orientationConstraint := range constraints.GetOrientationConstraint() {
 		topoConstraints = true
-		p.addOrientationConstraints(from, to, orientationConstraint)
+		// TODO RSDK-9224: Our proto for constraints does not allow the specification of which frames should be constrainted relative to
+		// which other frames. If there is only one goal specified, then we assume that the constraint is between the moving and goal frame.
+		err := p.addOrientationConstraints(fs, startCfg, from, to, orientationConstraint)
+		if err != nil {
+			return false, err
+		}
 	}
-	return topoConstraints
+	return topoConstraints, nil
 }
 
-func (p *plannerOptions) addLinearConstraints(from, to spatialmath.Pose, linConstraint LinearConstraint) {
+func (p *plannerOptions) addLinearConstraints(
+	fs referenceframe.FrameSystem,
+	startCfg map[string][]referenceframe.Input,
```

Comment:
Since `CreateAbsoluteLinearInterpolatingConstraintFS` is a public function, like its non-FS counterpart, it could be called from anywhere, with any parameters, and there is not a guarantee that PiFs in `from` have a counterpart in `to`, or if they do, that the parent frames of the PiFs is the same.

Passing in the framesystem and starting configuration allows the receiving function to calculate what it needs.

This wasn't previously necessary, because it was one pose to one pose, in the context of a specific frame, so there was no opportunity for inconsistent information.